test/request_reply_message_receiver_test.go

```diff
@@ -0,0 +1,123 @@
+// pubsubplus-go-client
+//
+// Copyright 2021-2024 Solace Corporation. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package test
+
+import (
+	"fmt"
+
+	"solace.dev/go/messaging"
+	"solace.dev/go/messaging/pkg/solace"
+	"solace.dev/go/messaging/pkg/solace/config"
+	"solace.dev/go/messaging/pkg/solace/resource"
+	"solace.dev/go/messaging/test/helpers"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+)
+
+const subscriptionString = "some-subscription"
+
+var subscription = resource.TopicSubscriptionOf(subscriptionString)
+
+var _ = Describe("RequestReplyReceiver", func() {
+	var messagingService solace.MessagingService
+	BeforeEach(func() {
+		var err error
+		messagingService, err = messaging.NewMessagingServiceBuilder().FromConfigurationProvider(helpers.DefaultConfiguration()).Build()
+		Expect(err).To(BeNil())
+	})
+
+	It("fails to build when given an invalid backpressure type", func() {
+		receiver, err := messagingService.RequestReply().CreateRequestReplyMessageReceiverBuilder().FromConfigurationProvider(config.ReceiverPropertyMap{
+			config.ReceiverPropertyDirectBackPressureStrategy: "not a strategy",
+		}).Build(subscription)
+		Expect(err).To(HaveOccurred())
+		Expect(err).To(BeAssignableToTypeOf(&solace.IllegalArgumentError{}))
+		Expect(receiver).To(BeNil())
+	})
+
+	It("fails to build when given an invalid subscription", func() {
+		badSubscription := &myCustomRequestReplySubscription{}
+		receiver, err := messagingService.RequestReply().CreateRequestReplyMessageReceiverBuilder().Build(badSubscription)
+		Expect(err).To(HaveOccurred())
+		Expect(err).To(BeAssignableToTypeOf(&solace.IllegalArgumentError{}))
+		Expect(err.Error()).To(ContainSubstring(fmt.Sprintf("%T", badSubscription)))
+		Expect(receiver).To(BeNil())
+	})
+
+	It("fails to start on unstarted messaging service", func() {
```

Comment:
Should this be named: "fails to start on unconnected messaging service"
Messaging service do not start they connect?

Reply:
yeah sure, it makes sense. I can rename it.